Added a learning mode

sevbesau opened the pull request:
I have added a new setting called "Show Full Algorithm".

Enabling this setting will replace the hint button with the full solution alg for the current case.

The reason for this addition is to make it easier to learn new cases. I am pretty new to advanced f2l, so don't know any of the algs yet. I found myself spamming the show hint button to reveal algs I did not know yet to learn them. 

This makes it so you can quickly go trough a set of selected cases and learn them.

Dave2ooo commented on scripts/main.js:
```diff
@@ -1376,3 +1380,13 @@ function openDialog(ELEM) {
   ELEM_BODY.style.overflow = "hidden";
 }
+
+function getCurrentAlgorithm() {
```
Is this function needed?

sevbesau replied:
No it is not, its left over from my experimenting. I have removed it!

Dave2ooo commented on scripts/main.js:
```diff
@@ -718,44 +733,35 @@ function generateTrainCaseList() {
 }
 
 function nextScramble(nextPrevious) {
-  if (hintSelection == 1) {
```
Line 721-723 show a loading spinner while a new F2L image is loaded.
This only appears if the setting "Hint Image" is set to "2D Image".
Without this feature and slow internet, there is no feedback to the user that a new case is being loaded.

sevbesau replied:
You're right! I have added this block of code again to address this issue. Complete oversight on my part!

I also noticed a bug on firefox when clicking on a select option in the settings modal. 
Clicking one of the options in the select window closed the whole modal without registering the new clicked value.
This prevented me from selecting the 2d visualization mode on firefox. 
I have also fixed this.

Dave2ooo commented on scripts/main.js:
```diff
@@ -718,44 +733,35 @@ function generateTrainCaseList() {
-    generateTrainCaseList();
-    flagSave = false;
-  }*/
-  if (nextPrevious) {
```
This new block of code breaks the functionality to revisit previous F2L cases.

The variable **nextPrevious** is 1 if the next case shall be shown and 0 if the previous case shall be shown.

Furthermore, if **currentTrainCaseNumber >= generatedScrambles.length** the function **generateTrainCaseList** shall be executed to generate more cases. This is to not repeat all cases in the same order like it would with **currentTrainCaseNumber = 0**

sevbesau replied:
You're right, this does indeed break some things like you say. I refactored this a bit to get more understanding into what is going on, but obviously missed some functionality. 

I have restored this block to its former state to fix the issue.

Dave2ooo reviewed (changes requested):
Hi, thanks for your commit, I really appreciate it.

I like the new feature and would approve it.
However, I have some questions/change requests - see comments.

sevbesau commented:
Thanks a lot for your feedback! 

I did indeed create some new bugs trying to get my bearings in the code. I have addressed the issues you highlighted and also fixed a firefox specific bug I encountered along the way!

Dave2ooo reviewed (approved):
Looks good. I'll approve the PR.
Thanks for your contribution.
Now I also know how to make a change request on GitHub.
